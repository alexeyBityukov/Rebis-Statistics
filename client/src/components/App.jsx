import React from 'react';
import { Provider } from 'react-redux';
import { MuiThemeProvider } from '@material-ui/core';
import store from '../store';
import theme from '../theme';
import renderRoutes from '../routers';

export default () => (
  <Provider store={store}>
    <MuiThemeProvider theme={theme}>
      {renderRoutes()}
    </MuiThemeProvider>
  </Provider>
);
