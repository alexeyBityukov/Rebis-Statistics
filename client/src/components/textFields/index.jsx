import React from 'react';
import { TextField } from '@material-ui/core';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


export const RenderTextField = ({
  input,
  label,
  meta: { touched, error },
  children,
  errors,
  required,
}) => (
  <TextField
    error={Boolean(touched && error)}
    helperText={Boolean(touched && error) && errors[error]}
    margin="normal"
    fullWidth
    required={required}
    label={label}
    {...input}
  >
    {children}
  </TextField>
);

RenderTextField.propTypes = {
  meta: PropTypes.shape({}),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  input: PropTypes.shape({}),
  label: PropTypes.string,
  errors: PropTypes.shape({}),
  required: PropTypes.bool,
};

RenderTextField.defaultProps = {
  meta: {},
  children: null,
  input: {},
  label: '',
  errors: {},
  required: false,
};

export default connect(state => ({ test: state.main.lang.errors }))(RenderTextField);
