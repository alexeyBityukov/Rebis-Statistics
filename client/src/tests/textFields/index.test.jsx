import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { RenderTextField } from '../../components/textFields';


configure({ adapter: new Adapter() });

const compareProps = (component, props) => {
  Object.keys(props).forEach((key) => {
    expect(component.prop(key)).toEqual(props[key]);
  });
};

describe('<RenderTextField>', () => {
  it('should render correctly', () => {
    const label = 'labelText';
    const component = shallow(<RenderTextField label={label} />);

    expect(shallowToJson(component)).toMatchSnapshot();
    compareProps(component, {
      label,
      error: false,
      fullWidth: true,
      helperText: false,
      margin: 'normal',
      required: false,
    });
  });

  it('should render with helperText', () => {
    const label = 'labelText';
    const helperText = 'error message';
    const errors = { errorKey: helperText };
    const component = shallow((
      <RenderTextField
        meta={{ touched: true, error: 'errorKey' }}
        errors={errors}
        required
        label={label}
      />));

    compareProps(component, {
      label,
      error: true,
      fullWidth: true,
      helperText,
      margin: 'normal',
      required: true,
    });

    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
