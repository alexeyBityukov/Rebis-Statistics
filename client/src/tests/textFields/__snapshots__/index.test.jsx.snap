// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderTextField> should render correctly 1`] = `
<TextField
  error={false}
  fullWidth={true}
  helperText={false}
  label="labelText"
  margin="normal"
  required={false}
  select={false}
  variant="standard"
/>
`;

exports[`<RenderTextField> should render with helperText 1`] = `
<TextField
  error={true}
  fullWidth={true}
  helperText="error message"
  label="labelText"
  margin="normal"
  required={true}
  select={false}
  variant="standard"
/>
`;
